lib/server/tcp-server.js

```diff
@@ -42,23 +43,24 @@ function Req(msg) {
 function Res(socket, ack) {
   this.ack = ack;
 
-  this.end = function() {
-    socket.write(VT + ack.toString() + FS + CR);
+  this.end = function(overrideAck) {
+    socket.write(VT + (overrideAck || ack).toString() + FS + CR);
   }
 }
 
-TcpServer.prototype.start = function(port, encoding) {
+TcpServer.prototype.start = function(port, encoding, options) {
   var self = this;
+  options = options || {}
   this.server = net.createServer(function(socket) {
     var message = "";
     socket.on('data', function(data) {
       try {
         message += data.toString();
-        if (message.substring(message.length - 2, message.length) == FS + CR) {
+        if (message.substring(message.length - 2, message.length) == FS + CR || options.skipValidation) {
```

Comment:
you will run into problems here if your message/messages ends up split between 2 data events. I've seen this happen on slow networks, or when the client sends hundreds of requests in quick succession.

you need someway of know when the message has ended, so you can buffer the data until you reach the end.

this seems like it could get tricky if the client is sending a stream of messages all at once, without waiting for a response. thats something you will probably have to work out with the client.

Reply:
Good point, I'll just remove this part